src/iris/api.py

```diff
@@ -3305,6 +3360,11 @@ def on_get(self, req, resp):
         if tag_subquery != "":
             where.append(tag_subquery)
 
+        tag_exists_subqueries = gen_tag_exists_where_subquery(connection, 'template_id', 'template_metadata_tag', '`template`.`id`', req.params)
+        if tag_exists_subqueries:
+            for tag_exists_subquery in tag_exists_subqueries:
+                where.append(tag_exists_subquery)
```

Comment:
I wonder if this could be simplified a bit if we created a new function that abstracts away both of the `gen_tag*` functions and returns a list of where clauses, so we can avoid maintaining both call sites 

Something like

```python
def gen_tag_wheres(...):
	wheres = []
	tag_subquery = gen_tag_where_subquery(...)
	if tag_subquery:
		wheres.append(tag_subquery)
	tag_exists_subqueries = gen_tag_exists_where_subquery(...)
	if tag_exists_subqueries:
		wheres.extend(tag_exists_subqueries)
    return wheres
```

And at call time in each of these uses, just:

```python
tag_wheres = gen_tag_wheres(connection, 'template_id', 'template_metadata_tag', '`template`.`id`', req.params)
if tag_wheres:
	where.extend(tag_wheres)
```

Reply:
makes sense to me, added